src/types/k8s.go

```diff
@@ -2,13 +2,30 @@ package types
 
 // ZarfState is maintained as a secret in the Zarf namespace to track Zarf init data
 type ZarfState struct {
-	ZarfAppliance bool         `json:"zarfAppliance"`
-	Distro        string       `json:"distro"`
-	Architecture  string       `json:"architecture"`
-	StorageClass  string       `json:"storageClass"`
+	ZarfAppliance bool         `json:"zarfAppliance" jsonschema:"description=Indicates if Zarf was initialized while deploying its own k8s cluster"`
+	Distro        string       `json:"distro" jsonschema:"description=K8s distribution of the cluster Zarf was deployed to"`
+	Architecture  string       `json:"architecture" jsonschema:"description=Machine architecture of the k8s node(s)"`
+	StorageClass  string       `json:"storageClass" jsonschema:"StorageClass of the k8s cluster Zarf was deployed to"`
 	Secret        string       `json:"secret"`
 	NodePort      string       `json:"nodePort"`
-	AgentTLS      GeneratedPKI `json:"agentTLS"`
+	AgentTLS      GeneratedPKI `json:"agentTLS" jsonschema:"PKI certificate information for the agent pods Zarf manages"`
+}
+
+type DeployedPackage struct {
+	PackageName string
+	PackageYaml ZarfPackage
+	CLIVersion  string
+
+	DeployedComponents map[string]DeployedComponent
+}
+
+type DeployedComponent struct {
+	InstalledCharts []InstalledCharts
```

Comment:
Do we care about images & repos too here?

Reply:
I didn't think so at first. All images/repos for a component is pushed to the registry/repository no matter what so since we have the struct of the package we already know what has been pushed.

Reply:
The main reason we have this list is since we turn raw manifests into Helm charts and I wanted a place to track what those chart names were.

Reply:
Hmm okay that can be a later feature I guess, but it seems reasonable to plan to store those as well like we are storing the charts (which also is in the zarf yaml)